test.py

```diff
@@ -574,9 +574,14 @@ def test(query, result):
         assert_equal(url.parse(query).pld, result)
 
     examples = [
-        ('http://foo.com/bar'    , 'foo.com'),
-        ('http://bar.foo.com/bar', 'foo.com'),
-        ('/foo'                  , '')
+        ('http://foo.com/bar'     , 'foo.com'),
+        ('http://bar.foo.com/bar' , 'foo.com'),
+        ('/foo'                   , ''),
+        ('http://com/bar'         , ''),
+        ('http://foo.გე'          , 'foo.გე'),
+        ('http://bar.foo.გე'      , 'foo.გე'),
+        ('http://foo.xn--node'    , 'foo.xn--node'),
+        ('http://bar.foo.xn--node', 'foo.xn--node')
```

Comment:
It doesn't look like any of your tests check for a non-trivial TLD like `.co.uk`.